core/src/main/java/io/questdb/cairo/vm/MemoryCARWSpillable.java

```diff
@@ -0,0 +1,283 @@
+package io.questdb.cairo.vm;
+
+import java.util.UUID;
+
+import org.jetbrains.annotations.NotNull;
+
+import io.questdb.cairo.CairoConfiguration;
+import io.questdb.cairo.vm.api.MemoryCARW;
+import io.questdb.log.Log;
+import io.questdb.log.LogFactory;
+import io.questdb.std.BinarySequence;
+import io.questdb.std.FilesFacade;
+import io.questdb.std.FilesFacadeImpl;
+import io.questdb.std.Long256;
+import io.questdb.std.MemoryTag;
+import io.questdb.std.Mutable;
+import io.questdb.std.Numbers;
+import io.questdb.std.Vect;
+import io.questdb.std.str.Path;
+
+public class MemoryCARWSpillable implements MemoryCARW, Mutable {
```

Comment:
This class is `FastMap` specific, so I'd move it to the `io.questdb.cairo.map` package and give it a more obvious name like `SpillableMapHeap` or `SpillableHeapMemory`.

Reply:
I see. I was thinking if it can be used more generally. But anyway, let me move if no other classes would need it.

Reply:
Introducing an abstraction preliminary, i.e. without a few use cases around it, usually doesn't end up with something maintainable. That's why it's better not to introduce another general-purpose memory class until we have a few use cases for it. ATM we only have one use case, so it's better to write something suitable for that use case.